src/hooks/cashout/usePrecalculatedCashouts.ts

```diff
@@ -0,0 +1,82 @@
+import { type Selection, GraphBetStatus } from '@azuro-org/toolkit'
+import { useMemo } from 'react'
+import { useQuery } from '@tanstack/react-query'
+
+import { useChain } from '../../contexts/chain'
+import { batchFetchCashouts } from '../../helpers/batchFetchCashouts'
+
+
+type Props = {
+  selections: Selection[]
+  graphBetStatus: GraphBetStatus
+  enabled?: boolean
+}
+
+export type PrecalculatedCashout = {
+  isAvailable: boolean
+  multiplier: string
+} & Omit<Selection, 'coreAddress'>
+
+
+export const usePrecalculatedCashouts = ({ selections, graphBetStatus, enabled = true }: Props) => {
+  const { appChain, api, contracts } = useChain()
+
+  const isLive = selections[0]!.coreAddress === contracts.liveCore?.address
+  const conditionsKey = useMemo(() => selections.map(({ conditionId }) => conditionId).join('-'), [ selections ])
+
+  const queryFn = async () => {
+    const data = await batchFetchCashouts(conditionsKey.split('-'), appChain.id)
+
+    const newCashouts = selections.reduce<Record<string, PrecalculatedCashout>>((acc, { conditionId, outcomeId }) => {
+      const key = `${conditionId}-${outcomeId}`
+      const cashout = data?.[key]!
+
+      acc[key] = cashout
+
+      return acc
+    }, {})
+
+    return newCashouts
+  }
+
+  const { data: cashouts, isFetching } = useQuery({
+    queryKey: [ 'cashout/precalculate', api, conditionsKey ],
+    queryFn,
+    gcTime: 0, // disable cache
+    refetchInterval: 60_000,
+    refetchOnWindowFocus: false,
+    enabled: (
+      enabled &&
+      Boolean(selections.length) &&
+      graphBetStatus === GraphBetStatus.Accepted &&
+      !isLive
+    ),
+  })
+
+  const isCashoutAvailable = useMemo(() => {
+    if (!cashouts || !Object.keys(cashouts).length) {
+      return false
+    }
+
+    return Object.values(cashouts).every(({ isAvailable }) => isAvailable)
```

Comment:
I'm a little confused. Does this hook accept only one bet or an array of bets?

Reply:
the query returns object for each selection, combo bet contains 2 or more selection. So we need to check every cashout